x-pack/plugins/code/server/repository_service.ts

```diff
@@ -47,6 +48,10 @@ export class RepositoryService {
           // move on with the clone.
           await this.remove(repo.uri);
         }
+      } else {
+        const parentDir = path.dirname(localPath);
+        // under windows, git clone will failed if parent folder is not exists;
```

Comment:
nit: `under` -> `on`. 

Also, try to avoid sync function because previously we found that it will block the server, to some extent.